### PR TITLE
MAINT: signal: malloc fail check

### PR DESCRIPTION
**Pull request by clementkng**
Added malloc checks for unchecked malloc calls based on issue #9999 in `sigtoolsmodule.c`

### CODE REVIEW

**Review by peterbell10 — COMMENTED** [review]
You will also have to edit the code after the `fail:` label since it doesn't `free` the `malloc`ed memory.

**Comment by clementkng** [comment]
@peterbell10 The easiest way to do that would be to just check each `malloc`ed variable equals `NULL` and if not, then free it right?

**Comment by peterbell10** [comment]
>The easiest way to do that would be to just check each malloced variable equals NULL and if not, then free it right?

Yes, except `free(NULL)` is a legal noop so I would just make sure the variables are initialised to NULL at the declaration and unconditionally `free`.

File: scipy/signal/sigtoolsmodule.c [hunk 1]
```diff
@@ -1011,6 +1015,15 @@ PyObject *PyArray_OrderFilterND(PyObject *op1, PyObject *op2, int order) {
 
 fail:
 	if (zptr) PyDataMem_FREE(zptr);
+	if (sort_buffer) free(sort_buffer);
```
Comment (pv) [inline scipy/signal/sigtoolsmodule.c]: The variables need to be explicitly initialized to NULL in the declarations at the beginning.

**Review by pv — CHANGES_REQUESTED** [review]

**Review by peterbell10 — APPROVED** [review]

**Pull request merged**